config/config_test.go

```diff
@@ -51,3 +54,43 @@ func TestNewSDK(t *testing.T) {
 		require.Equal(t, tt.wantShutdownErr, sdk.Shutdown(context.Background()))
 	}
 }
+
+func TestParseJSON(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		wantErr  error
+		wantType interface{}
+	}{
+		{
+			name:     "valid JSON",
+			input:    `{"file_format": "json", "disabled": false}`,
```

Comment:
This would return the same result

```suggestion
			input:    `{"file_format": "json", "disabled": false}THIS IS INTENTIONALLY MALFORMED`,
```

Reference: https://dev.to/taqkarim/you-might-not-be-using-json-decoder-correctly-in-golang-12mb

Maybe it would be safer to change the API to accept `[]byte` instead of `io.Reader` and use `json.Unmarshal`? Especially that the performance difference is not big. Reference: https://dev.to/jpoly1219/to-unmarshal-or-to-decode-json-processing-in-go-explained-5870

Or maybe we should even not add such function, but simply add an an example to docs instead?

E.g. 

```go
func ExampleParseJSON() {
	input := `{"file_format": "0.1", "disabled": true}`

	var cfg config.OpenTelemetryConfiguration
	if err := json.Unmarshal(([]byte)(input), &cfg); err != nil {
		panic(err)
	}

	fmt.Printf("Disabled: %v", cfg.Disabled)

	// Output: Disabled: true
}
```

I am leaning towards adding a runnable example in docs instead of adding a function.

Reply:
A runnable example in the docs sounds just fine to me. I can also rewrite this to accept the `[]byte` string so we can use the more dependable `json.Unmarshal` method in the parser if we decide to take this route.

Reply:
I am leaning towards a runnable a example to reduce the API surface.

When I created https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4412 I did NOT want `OpenTelemetryConfiguration` to have `mapstructure` tags, but after some time and disucssion agreed that it is acceptable. I will update the issue.

CC @codeboten